device/ethernet.yaml

```diff
@@ -1,58 +1,84 @@
-          type: array
-          items:
-            $ref: './ipv6.yaml#/components/schemas/Device.Ipv6'
-      
\ No newline at end of file
+components:
+  schemas:
+    Device.EthernetBase:
+      x-include: 
+      - '../common/common.yaml#/components/schemas/Named.Object'
+      description: >-
+        Base Ethernet interface.
+      type: object
+      required: [mac]
+      properties:
+        mac:
+          description: >-
+            Media Access Control address.
+          type: string
+          format: mac
+        mtu:
+          description: >-
+            Maximum Transmission Unit.
+          type: integer
+          minimum: 0
+          maximum: 65535
+          default: 1500
+        vlans:
+          description: >-
+            List of VLANs
+          type: array
+          items:
+            $ref: './vlan.yaml#/components/schemas/Device.Vlan'
+
+    Device.Ethernet:
+      x-include: 
+      - '#/components/schemas/Device.EthernetBase'
+      description: >-
+        An Ethernet interface with IPv4 and IPv6 addresses.
+      type: object
+      properties:
+        connection: 
+          description: |-
+            Device connection to physical, LAG or another device.
+          $ref : '#/components/schemas/Device.Connection'
+        port_name: 
+          description: |-
+            The unique name of a Port or a LAG that will emulate this interface.
+          type: string
+          x-constraint:
+          - '/components/schemas/Port/properties/name'
+          - '/components/schemas/Lag/properties/name'
+        ipv4_addresses:
+          description: |-
+            List of IPv4 addresses and their gateways. 
+          type: array
+          items:
+            $ref: './ipv4.yaml#/components/schemas/Device.Ipv4' 
+        ipv6_addresses:
+          description: |-
+            List of global IPv6 addresses and their gateways.
+            The Link Local IPv6 address will be automatically generated.
+          type: array
+          items:
+            $ref: './ipv6.yaml#/components/schemas/Device.Ipv6'
+      
+    Device.Connection:
+      description: Device connection to physical port, LAG port or another device.
+      type: object
+      properties:
+        choice:
+          description: port_name, lag_name or device_name
+          type: string
+          enum: [port_name, lag_name, device_name]
+        port_name:
+          description: Name of the port that the device is connected to.
+          type: string
+          x-constraint:
+          - '/components/schemas/Port/properties/name'
+        lag_name:
+          description: Name of the Lag the device is connected to.
+          type: string
+          x-constraint:
+          - '/components/schemas/Lag/properties/name'
+        device_name:
+          description: Name of the device that acts as the parent of this device.
+          type: string
+          x-constraint:
+          - 'components/schemas/Device/properties/name'

```

Comment:
Adding sub-set of my previous comments:
1. A device can have multiple L2 connectors. In reality we have to chain it to one of these L2 tunnels. Chaining to a device is hence ambiguous to actually implement. [ Unless we create an arbitrary implementation rule that it is always to be chained behind first L2 connector for example but that does not seem to scale ] . i.e. might want to connect to one of the many VxLAN interfaces in the parent DG or one of the L2GRE instances or one of the L2VPN instances. This does not seem to handle it. Other notes are mentioned in that comment. Chaining to a device using device name does not seem to suffice.

Reply:
Note that my comment is not related to port_name. I am talking about connection to a device. That itself looks incorrect to me. This connector is trying to connect to a particular instance of one of many types of L2 tunnels e.g. VXLAN, L2VPN or L2GRE

Reply:
yes got it. I'm thinking to update the models in the below way to have backward compatibility and based on your point that connection to a device doesn't make much sense. Please check and let me know your comments

1) Keep `port_name` as it is and remove required for now. Will add deprecated and remove it in future release.
2) `connection` will have three choices.
                     a) port_name
                     b) lag_name
                     c) l2_connector
3) As of now l2_connector will have x-constraint as vxlan name, will add l2gre&l2vpn in future based on user's requirement.

**Note:** We have to keep either `port_name` or `connection` required, will make connection as required once port_name is deprecated.

Sample test
========
```
# Create a VXLAN  device
vxlan = d1.vxlans.add()[-1]
vxlan.name = "VXLAN"

# Connected to VXLAN L2 Connector
e3 = d3.ethernets.ethernet()[-1]
e3.connection.l2_connector = vxlan.name              
e3.name = 'e3'
e3.mac = '00:01:00:00:00:03'
```

Reply:
Since port_name and lag_name have been broken out lets extend choice with vxlan_name and keep with specific descriptions?

Reply:
I think port_name and connection cannot be required until port_name is dropped from the model.

Reply:
> Since port_name and lag_name have been broken out lets extend choice with vxlan_name and keep with specific descriptions?

Agree with Andy.  Might be better to go with specific descriptions.

Reply:
updated as suggested